### PR TITLE
[WIP] SpaceTimeStack export data into JSON format

### PR DESCRIPTION
**Pull request by dalg24**
@pgrete

### CODE REVIEW

File: profiling/space-time-stack/kp_space_time_stack.cpp [hunk 1]
```diff
@@ -521,8 +604,19 @@ struct State {
...
-    auto inv_stack_root = stack_root.invert();
     stack_root.reduce_over_mpi();
+    if (getenv("KOKKOS_PROFILE_EXPORT_JSON")) {
+#if USE_MPI
+      int rank;
+      MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+      if (rank == 0)
+#endif
+        std::ofstream fout("noname.json");
+        stack_root.print_json(fout);
```
Comment (DavidPoliakoff) [inline profiling/space-time-stack/kp_space_time_stack.cpp:615]:
```suggestion
#if USE_MPI
      if (rank == 0) {
#endif
        std::ofstream fout("noname.json");
        stack_root.print_json(fout);
 #if USE_MPI
    }
 #endif
```

**Review by DavidPoliakoff — CHANGES_REQUESTED** [review]
Otherwise looks good

**Comment by DavidPoliakoff** [comment]
Actually, another bug with ExaMiniMD:

```
  120 {
  121 "average-time" : 4.38e-03,
  122 "percent" : 0.1,
  123 "percent-kokkos" : 199.5,
  124 "imbalance" : 0.0,
  125 "remainder" : "N/A",
  126 "kernels-per-second" : "N/A",
  127 "number-of-calls" : 39,
>>128 "name" : ""CommMPI::pack_count"="Scalar"",
  129 "parent-id" : "0xe4a370",
  130 "id" : "0xe4a410",
  131 "kernel-type" : "copy"
  132 },
```
Apparently something in how this string is quoted is breaking... all the joys of I/O libraries ;)

**Comment by nmhamster** [comment]
FYI - JSON double quotation marks inside a string must be escaped with ```\"```

**Comment by DavidPoliakoff** [comment]
@nmhamster : are you still allowed to do technical work? :p

**Comment by dalg24** [comment]
Fixed the double quote bug

**Comment by DavidPoliakoff** [comment]
@dalg24, I think my suggestion was broken in how it got applied, could you check that 
c24900a does what you want?

**Comment by dalg24** [comment]
> @dalg24, I think my suggestion was broken in how it got applied, could you check that
> [c24900a](https://github.com/kokkos/kokkos-tools/commit/c24900a194db223db62830296a9001f341c78a86) does what you want?

It does

**Review by DavidPoliakoff — APPROVED** [review]

**Pull request merged**